src/main/java/org/symphonyoss/symphony/messageml/elements/FormElement.java

```diff
@@ -34,4 +28,19 @@ public void assertParent(Collection<Class<? extends Element>> permittedParents)
           this.getMessageMLTag(), permittedParentsClassAsString));
     }
   }
+
```

Comment:
Still thinking about this _assertParent_ method, I see 3 options:

1. Have it overridden here, as it's now. This is my least favourite option as it introduces 2 different code paths to maintain; in addition, there is nothing specific to a Form that suggests that this method should be here, and not in the general Element class.

2. Don't override the method and leave the general _Element_ error message. This does force the API user to read the documentation, but, honestly, I'm OK with it.

3, Change the logic and error message in the general _Element_ class. I think it should be OK. This would be my second best option.

Reply:
I really think that just saying "cannot be a child of X" doesn't help much as it does not specify which elements it can be a child of, which would be much more informative. I'm inclined on option 3, I just overrode to not change the behavior for already supported tags.

Reply:
So I guess option 3 it is. It can be done in a separate PR to not delay this one any more.

Reply:
I did it on this one =P hahaha